### DISCUSSION TITLE
Suggestions to improve Thymeleaf: Very important

### DISCUSSION THREAD

**Discussion by slejmi** [thymeleaf/thymeleaf#982] [Ideas]
Dear People,
i am a software developer and trainer in Java SE/Java EE. I've been teaching Preparation Courses for all Oracle Java Certifications since 11 Years now. 

A few days ago, I startet to learn Thymeleaf, because after some research, it seems to be recommended for Server-Side Generated View.
Another Good Reason is the support from the Spring Projects, especially "Spring Boot".

Now After starting to learn a little bit, i am shocked why the syntax of some expressions is NOT INTUITIVE at all.

For example: 
Let's start with 
_______________________________________
Hyperlinks:
`<a th:href="@{/users}"></a>`
There are no Variables at all, why complicating things? Why not just use this: 
`<a th:href="/users"></a>`  i can still use `<a href="/...">` if i don't want context relative paths
_______________________________________
Request Parameters:
You use this Syntax:  `<a th:href="@{https://www.google.com/search(q=${searchTerm})"></a>`
Why not this:            `<a th:href="https://www.google.com/search?q=${serachTerm}"></a>`
_______________________________________
Path Variables: 
Use use: `<a th:href="@{/users/{userId}/edit(userId=${user.id})"></a>`   <== this one killed me :-)
Why not SIMPLY this: `<a th:href="/users/${user.id}/edit"></a>`
_______________________________________
String Concatenation:
You use this:   `<div th:id="|container-${index}|"></div>`
Why not this: `<div th:id="container-${index}"></div> `
_______________________________________
Expression Inlining:
Thymeleaf:      `<span>The total price is [[${totalPrice}]]</span>`
Why not just:  `<span>The total price is ${totalPrice}</span>`
_______________________________________
And i saw much more complicated Stuff in Thymeleaf.
My Recommendation: 
I don't know, if my recommendations will reduce the performance of your framework, but if it is not the case, please fix these non-intituive syntaxes asap.
Thank you very much.
Lejmi

**Comment by danielfernandez** [comment]
Thank you for your valuable contribution.
